src/post.ts

```diff
@@ -1,16 +1,31 @@
 import { setFailed } from '@actions/core'
-import { mkdirP, mv } from '@actions/io'
+import { mkdirP, mv, cp } from '@actions/io'
+import { spawn } from 'child_process'
+import { promisify } from 'util'
 
 import { getVars } from './lib/getVars'
 import { isErrorLike } from './lib/isErrorLike'
 import log from './lib/log'
 
+const rsync = promisify(spawn)
+
 async function post(): Promise<void> {
   try {
-    const { cacheDir, targetPath, cachePath } = getVars()
+    const { cacheDir, targetPath, cachePath, options } = getVars()
 
     await mkdirP(cacheDir)
-    await mv(targetPath, cachePath, { force: true })
+    if (options.copyStrategy === 'move') {
+      await mv(targetPath, cachePath, { force: true })
+    } else if (options.copyStrategy === 'copy') {
+      await cp(targetPath, cachePath, { copySourceDirectory: true, force: true, recursive: true })
```

Comment:
That's a great point.
Should that be configured with an additional flag? I feel that different cases could happen.

On that note, I think we should also have the option to not re-cache if there was a cache-hit. If you check the official [cache action](https://github.com/actions/cache), that's how it behaves by default (cache once).

A `copyStrategyPostType(?)` with something like `cache-once` (only caches if no cache hit), `replace` (always caches, deletes cache first), `overwrite(?)` would cover all cases.

Reply:
Nice findings!

Of course for `@actions/cache` it makes sense to make the cache immutable since the storage is remote, you don't want to incur in write costs on every job run.

So current implementation of `@masterworksio/action-local-cache` is mutable, and it is by design: as it is meant to be used with local, persisted storage, it benefits a lot from incremental caching.

I would say:

```
  strategy: 'move' | 'copy' | 'copy-immutable'
```